JarHellTaskIT failed to detect expected jar hell

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/13/consoleText failed with the following stack trace:

```
    BUILD FAILED in 12s
    8 actionable tasks: 8 executed
        at __randomizedtesting.SeedInfo.seed([E7EF87DDF5E1869C:4C9D378D4B52975A]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.elasticsearch.gradle.test.GradleIntegrationTestCase.assertOutputContains(GradleIntegrationTestCase.java:58)
        at org.elasticsearch.gradle.test.GradleIntegrationTestCase.assertOutputContains(GradleIntegrationTestCase.java:40)
        at org.elasticsearch.gradle.precommit.JarHellTaskIT.testJarHellDetected(JarHellTaskIT.java:35)
```

It looks like it's expecting to find `Exception in thread "main" java.lang.IllegalStateException: jar hell` but in fact got this:

```
    java.home: /var/lib/jenkins/.java/oracle-8u181-linux/jre
    examining directory: /var/lib/jenkins/workspace/elastic+elasticsearch+master+periodic/buildSrc/src/testKit/jarHell/build/classes/java/main
    examining jar: /var/lib/jenkins/workspace/elastic+elasticsearch+master+periodic/build/local-test-repo/org/elasticsearch/test/framework/7.0.0-alpha1-SNAPSHOT/framework-7.0.0-alpha1-20180916.174155-1.jar
    examining jar: /var/lib/jenkins/workspace/elastic+elasticsearch+master+periodic/build/local-test-repo/org/elasticsearch/elasticsearch/7.0.0-alpha1-SNAPSHOT/elasticsearch-7.0.0-alpha1-20180916.174153-1.jar
    Exception in thread "main" examining jar: /tmp/.gradle-test-kit-jenkins/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-api/2.11.1/268f0fe4df3eefe052b57c87ec48517d64fb2a10/log4j-api-2.11.1.jar
    java.lang.IllegalStateException: jar hell!
    class: org.apache.logging.log4j.Logger
    jar1: /var/lib/jenkins/workspace/elastic+elasticsearch+master+periodic/buildSrc/src/testKit/jarHell/build/classes/java/main
    jar2: /tmp/.gradle-test-kit-jenkins/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-api/2.11.1/268f0fe4df3eefe052b57c87ec48517d64fb2a10/log4j-api-2.11.1.jar
    	at org.elasticsearch.bootstrap.JarHell.checkClass(JarHell.java:277)
    	at org.elasticsearch.bootstrap.JarHell.checkJarHell(JarHell.java:190)
    	at org.elasticsearch.bootstrap.JarHell.checkJarHell(JarHell.java:86)
    	at org.elasticsearch.bootstrap.JarHell.main(JarHell.java:70)
```

I admit I'm a little puzzled that this now seems to be passing again.

--- a/buildSrc/src/test/java/org/elasticsearch/gradle/precommit/JarHellTaskIT.java
+++ b/buildSrc/src/test/java/org/elasticsearch/gradle/precommit/JarHellTaskIT.java
@@ -34,7 +34,7 @@ public class JarHellTaskIT extends GradleIntegrationTestCase {
         assertTaskFailed(result, ":jarHell");
         assertOutputContains(
             result.getOutput(),
-            "Exception in thread \"main\" java.lang.IllegalStateException: jar hell!",
+            "java.lang.IllegalStateException: jar hell!",
             "class: org.apache.logging.log4j.Logger"
         );
     }

